src/agent/EnvelopeService.ts

```diff
@@ -1,10 +1,19 @@
+import type { Verkey } from 'indy-sdk'
 import { inject, scoped, Lifecycle } from 'tsyringe'
-import { OutboundMessage, UnpackedMessageContext } from '../types'
+
+import { UnpackedMessageContext } from '../types'
 import { Wallet } from '../wallet/Wallet'
 import { ForwardMessage } from '../modules/routing/messages'
 import { AgentConfig } from './AgentConfig'
 import { Logger } from '../logger'
 import { Symbols } from '../symbols'
+import { AgentMessage } from './AgentMessage'
+
+interface Keys {
```

Comment:
I'd prefer a different name that Keys. Maybe something more verbose? 

`RecipientTarget`?

Reply:
Also I think it should be exported if it is going to be used by external services

Reply:
Hmm, I don't see `RecipientTarget` as more expressive 🤔  The interface represents a set of keys for packing a message. You can derive from context (it's used in `EnvelopeService` and `packMessage` method) the purpose of this object.

Right now, it seems it doesn't have to be exported. I don't like exporting stuff upfront. Or, can you give me an example of how you see the interface can be used externally?

Reply:
> Hmm, I don't see RecipientTarget as more expressive

I'm fine with another name, however keys is just so generic. It's not just keys. It's public keys used for didcomm messaging

Reply:
> Right now, it seems it doesn't have to be exported. I don't like exporting stuff upfront. Or, can you give me an example of how you see the interface can be used externally?

IMO all types used in public methods should be exported. As `packMessage` is public and uses the `Keys` type I think it should be exported. I find it very annoying when libraries have beautifully defined interfaces for public methods that I can't use (without a lot of typescript quirks) because they are exported. As the framework is extensible, I think it should be exported.

Maybe this is something we can solve with the monorepo:
- `@aries-framework/agent` -> all the normal public stuff. Does not export more than minimally required
- `@aries-framework/core` -> does export more than minimally needed, allows you to import types needed for extension